lib/common/json-schema-validator.js

```diff
@@ -176,19 +181,20 @@ function jsonSchemaValidatorFactory(
         // it search the schema object and replace the $ref object with the real value
         // example: { $ref: '#/definitions/test'} -> { type: 'string', format: 'uri'}
         function replaceRefObj (obj, resolvedValues, baseId, replaced) {
-            if (!(obj instanceof Object)) {
-                return;
-            }
-
+            // if found $ref key, other properties of the obj will be ignored
             if (obj.$ref) {
                 var refId = url.resolve(baseId, obj.$ref);
                 if (refId in resolvedValues) {
-                    replaced.push(refId);
+                    replaced[refId] = resolvedValues[refId];
                     return resolvedValues[refId];
                 }
             }
 
             for (var k in obj) {
```

Comment:
Good question. I know this maybe a concern. But the schema object is defined by ourselves, so it's surely that there will be no additional properties in the prototype chain.